src/python/pants/init/engine_initializer.py

```diff
@@ -217,7 +225,10 @@ def run_console_rules(self, options_bootstrapper, goals, target_roots):
         console.flush()
```

Comment:
For instance, I'm not convinced nothing is necessary to clean up the async reporter at this stage (but would handle in context manager as commented above)